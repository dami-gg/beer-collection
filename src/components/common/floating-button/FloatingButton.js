import React, {PureComponent} from 'react';

import './floating-button.scss';

class FloatingButton extends PureComponent {
  props: {
    iconClass: string,
    label: string,
    clickHandler: Function,
    buttonColor: string,
    iconColor: string
  };

  handleStyleCustomization() {
    let style = {};

    if (this.props.buttonColor) {
      style.backgroundColor = this.props.buttonColor;
    }
    if (this.props.iconColor) {
      style.color = this.props.iconColor;
    }

    return style;
  }

  render() {
    return (
      <div className="floating-button">
        <div onClick={this.props.clickHandler}
             className="floating-button__button"
             style={this.handleStyleCustomization()}>
          <i className={`${this.props.iconClass} floating-button__button__icon`}></i>
        </div>

        <div className="floating-button__label">
          <div className="floating-button__label__text">{this.props.label}</div>
          <i className="fa fa-play floating-button__label__arrow"></i>
        </div>
      </div>
    );
  }
}

export default FloatingButton;
