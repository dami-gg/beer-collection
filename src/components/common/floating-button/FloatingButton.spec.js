import React from "react";
import FloatingButton from "./FloatingButton";

import { shallow } from "enzyme";
import renderer from "react-test-renderer";

describe("FloatingButton", () => {
  let component, element, mock;

  beforeEach(mocks);

  it("should render without crashing", () => {
    shallow(<FloatingButton />);
  });

  it("should match the snapshot", () => {
    component = renderer.create(<FloatingButton />);
    let json = component.toJSON();

    expect(json).toMatchSnapshot();
  });

  describe("render", () => {
    beforeEach(setup);

    it("should render an element which on click calls the click handler", () => {
      element = component.find(".floating-button__button");
      element.simulate("click");

      expect(mock.props.clickHandler).toHaveBeenCalled();
    });

    it("should render an element with an icon with the class specified in props", () => {
      element = component.find(".floating-button__button");
      element = element.find(".floating-button__button__icon");

      expect(element.hasClass(mock.props.iconClass)).toBe(true);
    });

    it("should render an element with the label specified in props", () => {
      element = component.find(".floating-button__label__text");

      expect(element.text()).toBe(mock.props.label);
    });
  });

  function setup() {
    render();
  }

  function render() {
    component = shallow(
      <FloatingButton
        iconClass={mock.props.iconClass}
        label={mock.props.label}
        clickHandler={mock.props.clickHandler}
        buttonColor={mock.props.buttonColor}
        iconColor={mock.props.iconColor}
      />
    );
  }

  function mocks() {
    mock = {
      props: {
        iconClass: "class",
        label: "label",
        clickHandler: jest.fn(),
        buttonColor: "color",
        iconColor: "color"
      }
    };
  }
});
