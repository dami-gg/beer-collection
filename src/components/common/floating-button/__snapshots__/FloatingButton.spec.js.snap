// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FloatingButton should match the snapshot 1`] = `
<div
  className="floating-button">
  <div
    className="floating-button__button"
    onClick={undefined}
    style={Object {}}>
    <i
      className="undefined floating-button__button__icon" />
  </div>
  <div
    className="floating-button__label">
    <div
      className="floating-button__label__text" />
    <i
      className="fa fa-play floating-button__label__arrow" />
  </div>
</div>
`;
